packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -248,10 +304,79 @@ - (void)errorNoPhotoAccess:(PHAuthorizationStatus)status {
-  _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
-  [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
-                                                    animated:YES
-                                                  completion:nil];
+  if (__phPickerFlag) {
+    [[self viewControllerWithWindow:nil] presentViewController:__pickerViewController
+                                                      animated:YES
+                                                    completion:nil];
+  } else {
+    _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
+                                                      animated:YES
+                                                    completion:nil];
+  }
+}
+
+// Limited access to the photo library
+- (void)showLimitedPhotoLibrary {
+  [[PHPhotoLibrary sharedPhotoLibrary]
+      presentLimitedLibraryPickerFromViewController:__pickerViewController];
+}
+
+- (void)picker:(PHPickerViewController *)picker
+    didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+  [picker dismissViewControllerAnimated:YES completion:nil];
+
+  for (PHPickerResult *result in results) {
+    [result.itemProvider
+        loadObjectOfClass:[UIImage class]
+        completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
+                            NSError *_Nullable error) {
+          if ([image isKindOfClass:[UIImage class]]) {
+            if (image != nil) {
+              NSNumber *maxWidth = [self->__arguments objectForKey:@"maxWidth"];
+              NSNumber *maxHeight = [self->__arguments objectForKey:@"maxHeight"];
+              NSNumber *imageQuality = [self->__arguments objectForKey:@"imageQuality"];
+
+              if (![imageQuality isKindOfClass:[NSNumber class]]) {
+                imageQuality = @1;
+              } else if (imageQuality.intValue < 0 || imageQuality.intValue > 100) {
+                imageQuality = [NSNumber numberWithInt:1];
+              } else {
+                imageQuality = @([imageQuality floatValue] / 100);
+              }
+
+              if (maxWidth != (id)[NSNull null] || maxHeight != (id)[NSNull null]) {
+                image = [FLTImagePickerImageUtil scaledImage:image
+                                                    maxWidth:maxWidth
+                                                   maxHeight:maxHeight];
+              }
+
+              PHAsset *originalAsset =
+                  [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:result];
+
+              if (!originalAsset) {
+                // Image picked without an original asset (e.g. User took a photo directly)
+                [self saveImageWithPickerInfo:nil image:image imageQuality:imageQuality];
+              } else {
+                __weak typeof(self) weakSelf = self;
```

Comment:
Can put this line before the for loop so everything in the loop can use weakSelf

Reply:
Done!